torch/fx/passes/fake_tensor_prop.py

```diff
@@ -30,35 +30,45 @@ def __init__(self, module: torch.fx.GraphModule, mode: Optional[FakeTensorMode]
         self._mode = mode
 
     def run_node(self, n: Node):
-        import sympy
-        from torch.fx.experimental.symbolic_shapes import free_unbacked_symbols
 
         result = super().run_node(n)
-        sym = None
-        if (
-            'val' in n.meta and
-            isinstance(v := n.meta['val'], torch.SymInt) and
-            isinstance(v.node.expr, sympy.Symbol) and free_unbacked_symbols(v)
-        ):
-            sym = v
 
-        def extract_val(obj):
-            if isinstance(obj, FakeTensor):
-                return snapshot_fake(obj)
-            elif isinstance(obj, torch.Tensor):
-                # TODO: How is it possible that we get a non fake tensor?  We
-                # should be running under the mode...
-                return snapshot_fake(self._mode.from_tensor(obj, static_shapes=True))
-            elif isinstance(obj, py_sym_types):
-                return obj
+        nil = object()
+        scalar_types = (torch.SymInt, torch.SymFloat, torch.SymBool, int, float, bool)
+
+        def check_consistent(new, old=nil):
+            if isinstance(new, torch.Tensor):
+                if old is not nil:
+                    assert isinstance(old, torch.Tensor)
+                    torch._check(old.dim() == new.dim())
```

Comment:
Why is this a `torch._check` and not a plain `assert`? In my head, it would be a bug in the compiler somewhere if we were re-running fake tensor prop on our graph and one of the nodes' meta tensor values had a different number of dims.

(So far I've pictured `torch._check` mainly as an API for simplying dynamic shape equalities, but do we also use it in other places instead of plain assertions?)

Reply:
It's actually changing into rename_unbacked_to later. And yes, it is more assert-y than check-y, but I use check here because I need the other behavior it has (it's irrefutable, if there are unbacked symints, put it into deferred runtime asserts).